File: dehaze-tool/dehaze-sdk-android/src/main/java/com/pei/dehaze/sdk/model/algorithm/Algorithm.java
package com.pei.dehaze.sdk.model.algorithm;

import lombok.Data;
import java.util.List;

/**
 * 算法模型类
 */
@Data
public class Algorithm {
    private int id;
    private int parentId;
    private String name;
    private String type;
    private String description;
    private String img;
    private String path;
    private String importPath;
    private String params;
    private String flops;
    private Integer status;
    private String size;
    private List<Algorithm> children;
}

File: dehaze-go/internal/model/read/algorithm.go
package read

// Algorithm 算法读模型
type Algorithm struct {
	ID          int64       `json:"id"`
	ParentID    int64       `json:"parentId"`
	Name        string      `json:"name"`
	Type        string      `json:"type"`
	Img         string      `json:"img"`
	Description string      `json:"description"`
	Path        string      `json:"path"`
	Flops       string      `json:"flops"`
	Params      string      `json:"params"`
	ImportPath  string      `json:"importPath"`
	Status      int         `json:"status"`
	Size        string      `json:"size"`
	Children    []Algorithm `json:"children"`
}

func (a *Algorithm) GetID() int64 {
	return a.ID
}

func (a *Algorithm) GetParentID() int64 {
	return a.ParentID
}
